fabric-model-loading-api-v1/src/client/java/net/fabricmc/fabric/mixin/client/model/loading/ModelBakerMixin.java

```diff
@@ -76,6 +82,17 @@ private BiFunction<BlockState, BlockStateModel.UnbakedGrouped, BlockStateModel>
 		};
 	}
 
+	@ModifyReturnValue(method = "bake", at = @At("RETURN"))
+	private CompletableFuture<ModelBaker.BakedModels> withExtraModels(CompletableFuture<ModelBaker.BakedModels> models, @Local Executor executor, @Local ModelBaker.BakerImpl baker) {
+		if (fabric_eventDispatcher == null) return models;
+
+		CompletableFuture<Map<ExtraModelKey<?>, Object>> extraModels = AsyncHelper.mapValues(fabric_eventDispatcher.getExtraModels(), (id, model) -> model.bake(baker), executor);
```

Comment:
`(id, model) -> model.bake(baker)` should be replaced with
```java
(key, model) -> {
    try {
        return model.bake(baker);
    } catch (Exception e) {
        LOGGER.warn("Unable to bake extra model: '{}'", key, e);
        return null;
    }
}
```

If debug information is later added to the key, it can probably be included in the key's `toString` so this code won't need additional changes.